lib/queuePopulator/KafkaLogConsumer/LogConsumer.js

```diff
@@ -0,0 +1,169 @@
+const kafka = require('node-rdkafka');
+const async = require('async');
+const ListRecordStream = require('./ListRecordStream');
+
+// maximum time to wait for consumer group to rebalance (ms)
+const maxDelayRebalanceMs = 120000;
+
+class LogConsumer {
+
+    /**
+     * @constructor
+     * @param {Object} kafkaConfig queue populator kafka config
+     * @param {string} kafkaConfig.hosts kafka hosts
+     * @param {string} kafkaConfig.topic kafka oplog topic
+     * @param {string} kafkaConfig.consumerGroupId consumer group id
+     * @param {Logger} logger logger
+     */
+    constructor(kafkaConfig, logger) {
```

Comment:
@RachedBenMustapha @jonathan-gramain @alexanderchan-scality I know we follow this pattern all over the place but I was wondering if it was possible to instead of passing in configs, in this case for kafka but a similar pattern is used for mongo and zookeeper I believe, we pass in an already connected client. 
This would remove the need for the setup method as well as removing a responsibility we have. We also can reduce some of our code duplication.

Wouldn't block this PR cause it is following what we do but just mentioning it.